### PR TITLE
Handling invalid column names on kctrl list commands

### PR DESCRIPTION
**Pull request by rcmadhankumar**
#### What this PR does / why we need it:
Handling invalid column names on kctrl list commands.

**Sample Output:**
```
$ ./kctrl package available list --column=name,source,status,since,namespace                                         
kctrl: Error: invalid column names: source,status,since

$ ./kctrl package installed list --column=name,source,status,since,namespace                                         
kctrl: Error: invalid column names: source,since

$ ./kctrl package repository list --column=name,source,status,since,namespace                                        
kctrl: Error: invalid column names: since

$./kctrl app list --column=name,source,status,ss                                                                      
kctrl: Error: invalid column names: source,ss

$./kctrl package repo get -r repo  --column name,invalid                                    
kctrl: Error: invalid column names: invalid

$ ./kctrl app get -a dep-pkgi-9bl8ur -n kappctrl-e2e-ns-2 --column=name,ns,invalid     
kctrl: Error: invalid column names: ns,invalid

$./kctrl package available get -p pkg.parent.carvel.dev/1.0.0 -n kappctrl-e2e-ns-2 --column=ns,test
kctrl: Error: invalid column names: ns,test

$ ./kctrl package installed get -i dep-pkgi-d3xoyb -n kappctrl-e2e-ns-2 --column=name,ns,test 
kctrl: Error: invalid column names: ns,test

$ ./kctrl app status -a dep-pkgi-9bl8ur -n kappctrl-e2e-ns-2 --column=name,ns,invalid       
kctrl: Error: invalid column names: ns,invalid

$ ./kctrl package installed status -i dep-pkgi-d3xoyb -n kappctrl-e2e-ns-2 --column=name,ns,test
kctrl: Error: invalid column names: ns,test

```

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

##### Review Checklist:

- [ ] Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [ ] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [ ] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [ ] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by praveenrewar — COMMENTED** [review]
Can we also add a test for one of these commands?

**Comment by rcmadhankumar** [comment]
> Can we also add a test for one of these commands?

Tests are added.

File: cli/pkg/kctrl/cmd/app/list.go [hunk 1]
```diff
@@ -23,10 +23,11 @@ type ListOptions struct {
 
 	NamespaceFlags cmdcore.NamespaceFlags
 	AllNamespaces  bool
+	columns        *[]string
```
Comment (praveenrewar) [inline cli/pkg/kctrl/cmd/app/list.go]: nit: newline before

**Review by praveenrewar — COMMENTED** [review]
We use PrintTable in some other commands as well, do you think we should update those too since column flag is global?

File: cli/pkg/kctrl/cmd/app/list.go (hunk 1 above)
Comment (rcmadhankumar) [reply cli/pkg/kctrl/cmd/app/list.go]: fixed.

**Comment by rcmadhankumar** [comment]
> We use PrintTable in some other commands as well, do you think we should update those too since column flag is global?


Updated all the commands which are using printTable.

File: cli/pkg/kctrl/cmd/core/validate_columns.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package core
```
Comment (praveenrewar) [inline cli/pkg/kctrl/cmd/core/validate_columns.go]: Maybe the file name should be `print_table.go`?

**Review by praveenrewar — APPROVED** [review]
Other than the one comment, it looks good.

File: cli/pkg/kctrl/cmd/core/validate_columns.go (hunk 2 above)
Comment (rcmadhankumar) [reply cli/pkg/kctrl/cmd/core/validate_columns.go]: fixed it.

**Pull request merged**